### PR TITLE
Content crud

### PR DESCRIPTION
**Pull request by neymarjimoh**
- handles CRUD on learning content

### CODE REVIEW

**Comment by Nicanor008** [comment]
@neymarjimoh could you kindly share a screenshot of your postman POST.

File: src/api/v1/contents/content_controllers.js [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+import { InternalServerError, CustomError } from '../../../utils/customError'
+import responseHandler from '../../../utils/responseHandler'
+import validateDate from '../../../utils/checkDate'
+import isEmpty from '../../../utils/isEmpty'
+
+export async function createContent(req, res, next) {
+    try {
+        const userId = req.userData.id
+        const { category, topic, content, focusGroup, estimatedDuration, deadline } = req.body
+        const isdateValid = validateDate(dueDate)
```
Comment (Nicanor008) [inline src/api/v1/contents/content_controllers.js:12]: `dueDate` is not defined, shouldn't it be deadline?

**Review by Nicanor008 — CHANGES_REQUESTED** [review]
The update doesn't seem to work, response is 200 but nothing has been updated. Not sure of delete, just gives `Auth Failed`, but will retry. Progress is good @neymarjimoh, good job. Just fix the minor issues.

File: src/api/v1/contents/content_model.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        author: {
+            type: Schema.Types.ObjectId,
+            ref: 'User',
+            required: true,
+        },
+        focusGroup: {
+            type: String,
+            default: 'shoman',
+        },
+        category: stringRequired,
```
[the remark this answers is not in the document]
Comment (neymarjimoh) [reply src/api/v1/contents/content_model.js:19]: okay

File: src/api/v1/contents/content_controllers.js (hunk 1 above)
Comment (neymarjimoh) [reply src/api/v1/contents/content_controllers.js:12]: yeah,

**Comment by neymarjimoh** [comment]
> The update doesn't seem to work, response is 200 but nothing has been updated. Not sure of delete, just gives `Auth Failed`, but will retry. Progress is good @neymarjimoh, good job. Just fix the minor issues.

can i see a screnshot @Nicanor008

**Comment by Nicanor008** [comment]
> > The update doesn't seem to work, response is 200 but nothing has been updated. Not sure of delete, just gives `Auth Failed`, but will retry. Progress is good @neymarjimoh, good job. Just fix the minor issues.
> 
> can I see a screenshot @Nicanor008
@neymarjimoh This is Patch for a mentor user, gives 200 but doesn't update. 
Screenshot 2021-01-26 at 12 41 38

**Comment by neymarjimoh** [comment]
> > > The update doesn't seem to work, response is 200 but nothing has been updated. Not sure of delete, just gives `Auth Failed`, but will retry. Progress is good @neymarjimoh, good job. Just fix the minor issues.
> > 
> > 
> > can I see a screenshot @Nicanor008
> 
> @neymarjimoh This is Patch for a mentor user, gives 200 but doesn't update.
> Screenshot 2021-01-26 at 12 41 38

alright, please can you make a get request to see if its updated?

File: src/api/v1/contents/content_controllers.js [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+            query,
+            {
+                $set: updatedFields,
+            },
+            { new: true },
+        )
+        if (!updatedContent) {
+            return next(new CustomError(404, "Content with the ID doesn't exist or has been deleted"))
+        }
+        return responseHandler(res, 200, updatedContent, 'Task updated successfully')
```
Comment (Nicanor008) [inline src/api/v1/contents/content_controllers.js:134]: you only updating the deadline? you've defined `updatedFields` as an empty object but you not initializing with anything. The update should allow the author to update anything on the learning content they created. Kindly check up this function @neymarjimoh

**Review by Nicanor008 — CHANGES_REQUESTED** [review]

File: src/api/v1/contents/content_controllers.js (hunk 3 above)
Comment (neymarjimoh) [reply src/api/v1/contents/content_controllers.js:134]: yeah, noticed it 
a simple spread on the `req.body` object would have been used to initialize the `updatedFields`

File: src/api/v1/contents/index.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+/**
+ * view a learning content
+ * /api/v1/contents/:contentId - GET
+ */
+router.get('/contents/:contentId', isAuthenticated, getContent)
+
+/**
+ * delete content - DELETE by admin
+ */
+router.delete('/contents/:contentId',isAdmin, deleteContent)
```
Comment (Nicanor008) [inline src/api/v1/contents/index.js:31]: We should use soft delete for the creator(mento/admin) and hard delete for admin. For now, I'll merge this, but can be worked on later.

**Review by Nicanor008 — APPROVED** [review]

**Pull request merged**